src/materialui/inputs.tsx

```diff
@@ -100,8 +126,101 @@ export const FileView = withStyles(fileViewStyles)(({ classes, file, onClearForm
 	</Grid>
 ));
 
+class FileViewWithModal extends React.Component<FileViewProps> {
+	state = {
+        open: false
+    };
+
+    handleOpen = () => {
+        this.setState({ open: true });
+    };
+
+    handleClose = () => {
+        this.setState({ open: false });
+    };
+
+    handleState = () => {
+        if (this.state.open === true) {
+            this.setState({ open: false });
+        }
+	};
+	
+	render() {
+		const { classes, file, onClearForm, errors = [] } = this.props;
+		const DefaultComp = () => {
+			return (
+				<a className={`${classes.noDecoration} ${classes.link}`} href={file.url}>
+					<Typography variant="subtitle1" className={classes.fileName}>{file.name}</Typography>
+				</a>
+			)
+		};
+		const ModalPreview = () => {
+			return (
+				<a className={`${classes.noDecoration} ${classes.link}`} onClick={this.handleOpen}>
+					<Typography variant="subtitle1" className={classes.fileName}>{file.name}</Typography>
+				</a>
+			)
+		};
+		const isImage = (file.mimeType === "image/png" || file.mimeType === "image/jpg" || file.mimeType === "image/jpeg") ? true : false;
```

Comment:
image/jpg is an invalid mime type, dont confuse with jpg extension

Reply:
Thanks, remove it.